Evaluation/apply_training.py

```diff
@@ -109,6 +110,12 @@ def main(cfg: DictConfig) -> None:
         json_file.write(json.dumps(filemap_data, indent=4))
         json_file.truncate()
 
+    # create symlink to input file
```

Comment:
Idea with symlinking is a nice one, however there is already such a map (pred. file <-> input file) stored in form of a json file (`pred_input_filemap.json`), so I am not sure that there is a need to have an additional symlinked map. But that's good that you pointed out this use case, when inputs ("eval_data" folder) are provided explicitly to `evaluate_performance.py` as `path_to_input`, because it seems to be inconvenient and can be automated. 

Basically, `pred_input_filemap.json` is currently used only if `path_to_pred` contains mlflow artifacts. But it can be also used in the case when `path_to_target` points to mlflow artifacts too (while `path_to_pred=null`). This can incorporate our use case for evaluating WP-like models, when previously we would specify:
* path_to_input = "eval_data/{sample_alias}/*.root"
* path_to_pred = null
* path_to_target = \<mlflow artifacts of some evaluated model\>

This approach is actually error-prone since the user should make sure that provided inputs correspond to the targets. So I'd suggest to correct this use case with setting `path_to_input=null` and automatically fetching it from `pred_input_filemap.json` similarly as it is currently done for `path_to_pred`, if `path_to_target` points to some mlflow run and the json exists there (otherwise raising an exception). This requires minor rearrangements in `prepare_filelists()` because currently if both `path_to_input` and `path_to_pred` are null, it will raise an assertion.